Emby.Server.Implementations/Session/SessionManager.cs

```diff
@@ -578,6 +582,66 @@ private void StopIdleCheckTimer()
             }
         }
 
+        private async Task CleanupDanglingStreams()
+        {
+            List<KeyValuePair<string, ILiveStream>> openStreams = _mediaSourceManager.GetOpenStreams();
+            List<bool> isStreamActive = new List<bool>(openStreams.Count);
+            isStreamActive.AddRange(Enumerable.Repeat(false, openStreams.Count));
+
+            // Do initial check to find any unclaimed streams
+            int streamIdx = 0;
+            foreach (KeyValuePair<string, ILiveStream> liveStream in openStreams)
+            {
+                if (liveStream.Value.AllowCleanup)
+                {
+                    isStreamActive[streamIdx] = false;
+                    foreach (SessionInfo session in Sessions)
+                    {
+                        if (!string.IsNullOrEmpty(session.PlayState?.LiveStreamId) && session.PlayState.LiveStreamId == liveStream.Key)
+                        {
+                            isStreamActive[streamIdx] = true;
+                            break;
+                        }
+                    }
+                }
+
+                streamIdx++;
+            }
+
+            await Task.Delay(10000);
+
+            // Check again and close inactive streams
+            streamIdx = 0;
+            foreach (KeyValuePair<string, ILiveStream> liveStream in openStreams)
+            {
+                if (liveStream.Value.AllowCleanup)
+                {
+                    foreach (SessionInfo session in Sessions)
+                    {
+                        if (!string.IsNullOrEmpty(session.PlayState?.LiveStreamId) && session.PlayState.LiveStreamId == liveStream.Key)
+                        {
+                            isStreamActive[streamIdx] = true;
+                            break;
+                        }
+                    }
+
+                    if (!isStreamActive[streamIdx])
+                    {
+                        try
+                        {
+                            await _mediaSourceManager.CloseLiveStream(liveStream.Key).ConfigureAwait(false);
```

Comment:
Would be better to queue up all the `CloseLiveStream` returned Task and then Await them all after the loop, or even not await them at all since this is really a fire-and-forget operation.